src/server.rs

```diff
@@ -1,39 +1,199 @@
+use crate::command::ClientCommand;
 use crate::{command::AOMessageCodec, config::Config};
-use tokio::net::TcpListener;
-use tokio_util::codec::Decoder as _;
+use bb8::Pool;
+use bb8_postgres::PostgresConnectionManager;
+use bytes::{BufMut, BytesMut};
+use futures::stream::SplitSink;
+use futures::SinkExt;
+use std::fmt::Debug;
+use tokio::net::{TcpListener, TcpStream};
+use tokio_postgres::NoTls;
+use tokio_util::codec::{Decoder, Encoder, Framed};
+
+#[rustfmt::skip]
+#[derive(Debug)]
+pub enum ServerCommand {
+    Handshake(String)
+}
+
+impl ServerCommand {
+    pub fn extract_args(&self) -> Option<Vec<&str>> {
```

Comment:
I propose moving this methods to `Command` trait and replacing `Option<Vec<&str>>` with `Vec<&str>`